iaf-commons/src/main/java/nl/nn/adapterframework/util/AdditionalStringResolver.java

```diff
@@ -0,0 +1,65 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.util;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.Set;
+
+/**
+ * Interface to be implemented by framework-classes that provide extra property-resolving sources, such as credentials, which cannot be directly included of the
+ * {@link StringResolver} methods.
+ * <p>
+ * Implementors of the interface should have a no-args constructor and not need any injected dependencies. The {@link StringResolver} will create instances
+ * of each class that implements this interface, and invoke it from {@link StringResolver#substVars(String, Map, Map, Set, String, String, boolean)}.
+ * </p>
+ * <p>
+ *     Implementors are loaded via the JDK {@link java.util.ServiceLoader} mechanism.
+ * </p>
+ * <p>
+ *  It is <b>not</b> allowed to use a static reference to the logger in implementations.
+ * 	Log4j2 uses StringResolver during instantiation.
+ * </p>
+ */
+public interface AdditionalStringResolver {
+	/**
+	 * Static field to access all instances via the {@link ServiceLoader}.
+	 */
+	ServiceLoader<AdditionalStringResolver> serviceLoader = ServiceLoader.load(AdditionalStringResolver.class);
```

Comment:
Maar deze is niet static?

Reply:
Brainfart.
It was a static field originally in the `StringResolver` class, then I moved it to the interface and then of course it is no longer static as it's part of the interface.

Still considering if it should be moved back to `StringResolver` though.